samza-core/src/main/java/org/apache/samza/clustermanager/ContainerAllocator.java

```diff
@@ -233,14 +231,15 @@ void assignResourceRequests() {
   }
 
   /**
-   * Handles an expired resource request when {@code hostAffinityEnabled} is true, in this case since the
-   * preferred host, we try to see if a surplus ANY_HOST is available in the request queue.
+   * Handles an expired resource request for both active and standby containers. Since a preferred host cannot be obtained
+   * this method checks the availability of surplus ANY_HOST resources and launches the container if available. Otherwise
+   * issues an ANY_HOST request. This behavior holds regardless of host-affinity enabled or not.
    */
   @VisibleForTesting
-  void handleExpiredRequestWithHostAffinityEnabled(String processorId, String preferredHost,
+  void handleExpiredRequest(String processorId, String preferredHost,
       SamzaResourceRequest request) {
     boolean resourceAvailableOnAnyHost = hasAllocatedResource(ResourceRequestState.ANY_HOST);
-    if (standbyContainerManager.isPresent()) {
+    if (hostAffinityEnabled && standbyContainerManager.isPresent()) {
```

Comment:
Don't need the hostAffinityEnabled check here.
You can assume standbyContainer is present only when host-affinity is enabled.

Reply:
sure